File: BlackFireIM/Example/Lib/MSIMSDK-UIKit/ChatDetail/View/Message/MSTextMessageCell.m
//
//  MSTextMessageCell.m
//  BlackFireIM
//
//  Created by benny wang on 2021/3/5.
//

#import "MSTextMessageCell.h"

@implementation MSTextMessageCell

- (instancetype)initWithStyle:(UITableViewCellStyle)style reuseIdentifier:(NSString *)reuseIdentifier
{
    self = [super initWithStyle:style reuseIdentifier:reuseIdentifier];
    if (self) {
        _content = [[UILabel alloc] init];
        _content.numberOfLines = 0;
        [self.bubbleView addSubview:_content];
    }
    return self;
}

- (MSTextMessageCellData *)textData
{
    return (MSTextMessageCellData *)self.messageData;
}

- (void)fillWithData:(MSTextMessageCellData *)data
{
    [super fillWithData:data];
    self.content.attributedText = data.attributedString;
    self.content.textColor = data.textColor;
}

- (void)layoutSubviews
{
    [super layoutSubviews];
    self.content.frame = (CGRect){.origin = self.textData.textOrigin, .size = self.textData.textSize};
}

@end

File: Swift/MSIM-Swift/Lib/MSIMSDK-UIKit/ChatDetail/View/Message/MSTextMessageCell.swift
//
//  MSTextMessageCell.swift
//  MSIM-Swift
//
//  Created by benny wang on 2021/8/6.
//

import UIKit

open class MSTextMessageCell: MSBubbleMessageCell {

    
    public var content: UILabel!
    
    public var textData: MSTextMessageCellData?
    
    public override init(style: UITableViewCell.CellStyle, reuseIdentifier: String?) {
        super.init(style: style, reuseIdentifier: reuseIdentifier)
        
        content = UILabel()
        content.numberOfLines = 0
        bubbleView.addSubview(content)
    }
    
    required public init?(coder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
    public override func fillWithData(data: MSMessageCellData) {
        super.fillWithData(data: data)
        if let textCellData = data as? MSTextMessageCellData {
            self.textData = textCellData
            content.attributedText = textCellData.attributedString
            content.textColor = textCellData.textColor
        }
    }
    
    open override func layoutSubviews() {
        super.layoutSubviews()
        if textData != nil {
            content.frame = CGRect(origin: textData!.textOrigin, size: textData!.textSize)
        }
    }
}
